From 7f3470d28dccc5a18c331d58730ea72d5bc27d5f Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Tue, 30 Apr 2013 14:21:52 +0800
Subject: libstagefright: A fix to ensure we can get the closest sync frame to the seek point

BZ: 55341

JB adopts binary search for the closest sync frame
to the seek point. We need to have the left end
subtract 1 to meet the need.

Category: aosp improvement
Domain: Video.Media-local playback
Origin: internal
Upstream-Candidate: yes

Change-Id: If21c15ebe060d3f9d0b9fded58bf49e47dcfb8ad
Orig-Change-Id: I67eca217ceb4b38b1b6cd0f8b92d5316809aa28e
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/SampleTable.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/SampleTable.cpp b/media/libstagefright/SampleTable.cpp
index 5f1bcd7..b560d84 100644
--- a/media/libstagefright/SampleTable.cpp
+++ b/media/libstagefright/SampleTable.cpp
@@ -644,6 +644,10 @@ status_t SampleTable::findSyncSampleNear(
         left = left - 1;
     }
 
+    if (left > 0) {
+        --left;
+    }
+
     // Now ssi[left] is the sync sample index just before (or at)
     // start_sample_index.
     // Also start_sample_index < ssi[left + 1], if left + 1 < mNumSyncSamples.
-- 
1.7.4.1

